lib/src/config/deprecated_option.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:meta/meta.dart';
+
+/// Class representing deprecated config option
+@immutable
+class DeprecatedOption {
+  /// Version number until which we support obsolete option
+  final String supportUntilVersion;
+
+  /// Obsolete option name
+  final String deprecated;
```

Comment:
Obsolete option name -> The deprecated option name. And I'd suggest to rename the field to `name`.

Reply:
fixed